src/api/general-services/pipeline-manage/constructors/assign-pod-to-pipeline.js

```diff
@@ -0,0 +1,107 @@
+const config = require('../../../../config');
+
+// the full activityArn is too long to be used as a tag (> 63 chars)
+// so we just send the last part of the arn as the rest can be constructed.
+//  E.g.
+// arn:aws:states:eu-west-1:242905224710:activity:pipeline-production-01037a63-a801-4ea4-a93e-...
+// => pipeline-production-01037a63-a801-4ea4-a93e-def76c1e5bd2
+const getActivityId = (activityArn) => {
+  if (activityArn === undefined) {
+    return undefined;
+  }
+
+  const split = activityArn.split(':');
+  return split[split.length - 1];
+};
+
+
+const assignPodToPipeline = (context, step) => {
+  const {
+    clusterInfo, activityArn, experimentId, processName,
+  } = context;
+
+
+  return {
+    ...step,
+    Type: 'Map',
+    ResultPath: null,
+    ItemsPath: '$.retries',
+    MaxConcurrency: 1,
+    // retry waits up to 226 seconds, fargate takes from 1 to 3 minutes to spawn a new pod
+    Retry: [{
+      ErrorEquals: ['NoPodsAvailable'],
+      IntervalSeconds: 2,
+      MaxAttempts: 11,
+      BackoffRate: 1.5,
+    }],
+    Iterator: {
+      StartAt: 'GetUnassignedPod',
```

Comment:
In the case there is temporarily not an available and unassigned pod, could this step fail? In that case do we retry?